functorch/csrc/BatchingRegistrations.cpp

```diff
@@ -448,40 +448,83 @@ Tensor& zero_inplace_batching_rule(Tensor &self) {
   return self;
 }
 
-Tensor squeeze_batching_rule(const Tensor& self) {
-  if (!participatesInCurrentLevel(self)) {
-    c10::impl::ExcludeDispatchKeyGuard guard(kBatchedKey);
-    return self.squeeze();
-  }
-  auto self_physical = MultiBatchVmapTransform::logicalToPhysical(self);
-  auto physical_sizes = self_physical.tensor().sizes();
-
-  // Don't squeeze the batch dims!
-  VmapDimVector squeezed_sizes;
-  int64_t num_batch_dims = self_physical.numBatchDims();
-  squeezed_sizes.insert(
-      squeezed_sizes.end(),
-      physical_sizes.begin(),
-      physical_sizes.begin() + num_batch_dims);
-  for (auto it = physical_sizes.begin() + num_batch_dims; it != physical_sizes.end(); ++it) {
-    if (*it != 1) {
-      squeezed_sizes.push_back(*it);
+std::tuple<Tensor, optional<int64_t>> squeeze_batching_rule(const Tensor& self, optional<int64_t> bdim) {
+  if (bdim) {
```

Comment:
I forgot to mention this, but `bdim` is never nullopt in an operator that accepts a single Tensor. If bdim is nullopt, then the batching rule doesn't actually get invoked; PyTorch dispatcher ends up not dispatching to the Batched implementation of things. Because of this we can add `TORCH_INTERNAL_ASSERT(bdim.has_value())` and delete the branch where bdim is nullopt.